.github/workflows/build-push-docker.yml

```diff
@@ -3,78 +3,18 @@ name: Build and Push Subgraph Deployer Docker Image
+      - '*'
+  pull_request:
+    branches:
+      - '*'
   workflow_dispatch:
- 
+
 jobs:
-  # TODO: See how to ensure that the default CI workflow has run before this workflow
   build-and-push:
-    if: |
-      (github.event_name == 'workflow_dispatch' && (
-        github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/heads/feature/')
-      )) || startsWith(github.ref, 'refs/tags/v')
-    runs-on: ubuntu-latest
-    env:
-      IS_TAG_TRIGGER: ${{ startsWith(github.ref, 'refs/tags/v') }}
-    steps:
-      - name: Get branch names
-        id: branch-names
-        uses: tj-actions/branch-names@v8
-
-      # Verify tag is on main or release branch
-      - name: Verify tag is on main or release branch
-        if: env.IS_TAG_TRIGGER == 'true'
-        run: |
-          if [[ "${{ steps.branch-names.outputs.base_ref_branch }}" == "main" || "${{ steps.branch-names.outputs.base_ref_branch }}" =~ ^release/ ]]; then
-            echo "✅ Tag ${{ steps.branch-names.outputs.tag }} is on an allowed branch: ${{ steps.branch-names.outputs.base_ref_branch }}"
-          else
-            echo "Current base branch: ${{ steps.branch-names.outputs.base_ref_branch }}"
-            echo "Tags must be created on main or release/* branches."
-            exit 1
-          fi
-
-      # Run npm version check only on main & release/**
-      - name: Verify npm package version matches tag
-        if: env.IS_TAG_TRIGGER == 'true'
-        uses: nick-y-ito/gha-npm-version-match@v1
-    
-      - name: Checkout Repository
-        uses: actions/checkout@v4
-
-      - name: Determine Docker Image Tags
-        run: |
-          IMAGE_NAME="iexechub/voucher-subgraph-deployer"
-          # Initialize IMAGE_TAG variable
-          IMAGE_TAG=""
-          
-          if [[ "$IS_TAG_TRIGGER" == "true" ]]; then
-            TAG_VERSION=${GITHUB_REF#refs/tags/v}
-            IMAGE_TAG="$IMAGE_NAME:$TAG_VERSION"
-          else
-            SHORT_COMMIT=$(git rev-parse --short HEAD)
-            IMAGE_TAG="$IMAGE_NAME:dev-$SHORT_COMMIT"
-          fi
-
-          echo "IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV
-          echo "✅ Resolved IMAGE_TAG: $IMAGE_TAG"
-
-      - name: Login to Docker Hub
-        uses: docker/login-action@v3
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_PAT }}
-
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-
-      - name: Build and Push Docker Image
-        uses: docker/build-push-action@v6
-        with:
-          file: docker/Dockerfile
-          push: true
-          platforms: linux/amd64,linux/arm64
-          tags: ${{ env.IMAGE_TAG }}
+    uses: iExecBlockchainComputing/github-actions-workflows/.github/workflows/docker-build.yml@docker-build-v1.1.1
+    with:
+      dockerfile: 'Dockerfile'
+      image-name: 'iexecdev/subgraph-deployer'
```

Comment:
1. Why iexecdev?
2. Why an image name without `voucher` in it?
3. Where do we make sure we are not erasing a existing image?

Reply:
Mistake, tags are the trust. And on branch I want to add an option to force the release with the unique sha of commit and branch name, only for debug.